application/src/main/java/com/sanction/lightning/facebook/FacebookService.java

```diff
@@ -102,17 +104,30 @@ public FacebookUser getFacebookUser() {
    *
    * @return A boolean to denote that the function worked.
    */
-  public FacebookPhoto publishToFacebook(byte[] inputBytes, String fileName, String message) {
-    FacebookPhoto response;
+  public String publishToFacebook(byte[] inputBytes, String type, String fileName,
+                                  String message, String videoTitle) {
+    JsonObject response;
+    List<Parameter> parameters = Lists.newArrayList();
+
+    if (type.equals("photo")) {
+      type = "photos";
+      parameters.add(Parameter.with("message", message));
+    } else if (type.equals("video")) {
+      type = "videos";
```

Comment:
Instead of doing this, can you create a local variable like `String apiEndpoint = "me/" + type + "s"`, and then use that `apiEndpoint` in the `client.publish()` method? Idk if that's any better than this or not.. let me know what you think.